app/src/main/java/cm/aptoide/pt/home/apps/SeeMoreAppcManager.java

```diff
@@ -0,0 +1,116 @@
+    this.updatesManager = updatesManager;
+    this.installManager = installManager;
+    this.appMapper = appMapper;
+    this.downloadFactory = downloadFactory;
+    this.downloadAnalytics = downloadAnalytics;
+    this.installAnalytics = installAnalytics;
+  }
+
+  public Observable<List<App>> getAppcUpgradesList(boolean isExcluded) {
+    return updatesManager.getAppcUpgradesList(isExcluded)
+        .distinctUntilChanged()
+        .map(updates -> appMapper.mapUpdateToUpdateAppList(updates));
+  }
+
+  public Observable<List<App>> getAppcUpgradeDownloadsList() {
+    return installManager.getInstallations()
+        .distinctUntilChanged()
+        .throttleLast(200, TimeUnit.MILLISECONDS)
+        .flatMap(installations -> {
+          if (installations == null || installations.isEmpty()) {
+            return Observable.empty();
+          }
+          return Observable.just(installations)
+              .flatMapIterable(installs -> installs)
+              .flatMap(install -> updatesManager.filterNonAppcUpgrade(install))
+              .toList()
+              .map(updatesList -> appMapper.getUpdatesList(updatesList, true));
+        });
+  }
+
+  public Completable refreshAllUpdates() {
+    return updatesManager.refreshUpdates();
+  }
+
+  public boolean showWarning() {
+    return installManager.showWarning();
+  }
+
+  public void storeRootAnswer(boolean answer) {
+    installManager.rootInstallAllowed(answer);
+  }
+
+  public Completable pauseDownload(App app) {
+    return Completable.fromAction(
+        () -> installManager.stopInstallation(((DownloadApp) app).getMd5()));
+  }
+
+  public Completable resumeUpdate(App app) {
+    return installManager.getDownload(((UpdateApp) app).getMd5())
+        .flatMapCompletable(download -> installManager.install(download));
+  }
+
+  public void cancelUpdate(App app) {
+    installManager.removeInstallationFile(((UpdateApp) app).getMd5(),
+        ((UpdateApp) app).getPackageName(), ((UpdateApp) app).getVersionCode());
+  }
+
+  public Completable pauseUpdate(App app) {
+    return Completable.fromAction(
+        () -> installManager.stopInstallation(((UpdateApp) app).getMd5()));
+  }
+
+  public Completable updateApp(App app) {
+    String packageName = ((UpdateApp) app).getPackageName();
+    return updatesManager.getUpdate(packageName)
+        .flatMap(update -> {
+          Download value = downloadFactory.create(update, true);
+          return Observable.just(value);
+        })
+        .doOnNext(download -> {
+          setupUpdateEvents(download, Origin.UPDATE, NO_ADS);
+        })
+        .flatMapCompletable(download -> installManager.install(download, true))
+        .toCompletable();
+  }
+
+  private void setupUpdateEvents(Download download, Origin origin,
+      WalletAdsOfferManager.OfferResponseStatus offerResponseStatus) {
+    downloadAnalytics.downloadStartEvent(download, AnalyticsManager.Action.CLICK,
+        DownloadAnalytics.AppContext.APPS_FRAGMENT);
```

Comment:
Maybe we should create a new AppContext called see more fragment so that we could distinguish the events coming from this view and the ones from Apps. But it is better to confirm with Ana